### DISCUSSION TITLE
Change Extension

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#116] [Share]
[Install change-extension](https://scriptkit.app/api/new?name=change-extension&url=https://gist.githubusercontent.com/johnlindquist/244216e5192eaa50f0c3089c0131d984/raw/120e17ef7debb9acb0e72aa76ef3e12cc5069b79/change-extension)

```js
// Menu: Change Extension
// Description: Drag and drop files to change their extension
// Author: John Lindquist
// Twitter: @johnlindquist

/** @type {File[]} */
let droppedFiles = await arg("Drop files")

let extension = await arg({
  placeholder: "Extension",
  hint: "ts, js, mp4, etc",
})

for await (let { path } of droppedFiles) {
  let tsPath = path.replace(/[^.]*$/, extension)
  mv(path, tsPath)
}
```

**Comment by zzeleznick** [comment]
Nice! I'm tempted to remix this to copy the finder UI for batch renaming. 

<img width="488" alt="File renames in finder" src="https://user-images.githubusercontent.com/5779832/113800755-5e27cb00-970c-11eb-80d5-0d88ebffbb09.png">

I love the little green visual accent and placeholder change onDragEnter!

One minor callout here is that the main window appears to accepts files by default. I dragged a script on it to test what would happen, and currently the prompts shows the error `input.replace is not a function` before exiting.

<img width="546" alt="Kit main window accepts file input" src="https://user-images.githubusercontent.com/5779832/113800819-77c91280-970c-11eb-81e4-83aa8eb7bdab.png">

**Reply by johnlindquist** [reply]
Yeah, I need to add a "drop" prop to to the promptConfig so it only enables when necessary.

I'm still in the middle of the TypeScript conversion for the kit repo, hoping to finish it up tomorrow, then I can get back to features.
